Modules/Core/Common/include/itkByteSwapper.hxx

```diff
@@ -337,6 +307,27 @@ ByteSwapper<T>::SwapBytes(T & value)
   }
 }
 
+template <typename T>
+void
+ByteSwapper<T>::SwapWriteRange(const T * buffer, SizeValueType numberOfElements, std::ostream & outputStream)
+{
+  auto chunkSize = std::min(numberOfElements, SizeValueType{ 1000000 });
+
+  const auto chunk = make_unique_for_overwrite<T[]>(chunkSize);
+
+  while (numberOfElements > 0)
+  {
+    std::copy_n(buffer, chunkSize, chunk.get());
+    std::for_each_n(chunk.get(), chunkSize, [](T & element) { SwapBytes(element); });
+
+    outputStream.write(reinterpret_cast<const char *>(chunk.get()),
+                       static_cast<std::streamsize>(chunkSize * sizeof(T)));
+    buffer += chunkSize;
+    numberOfElements -= chunkSize;
+    chunkSize = std::min(numberOfElements, chunkSize);
+  }
+}
```

Comment:
For comparison, down here is the existing `ByteSwapper<T>::SwapWrite8Range`, from the current master revision. (The existing `SwapWrite8Range` is very similar to `SwapWrite2Range` and `SwapWrite4Range` as well).
https://github.com/InsightSoftwareConsortium/ITK/blob/3b04ad1b377c6d5d77a2c1f454c363eab8197340/Modules/Core/Common/include/itkByteSwapper.hxx#L289-L312